make-prs.py

```
@@ -0,0 +1,251 @@
+
+    head = f"{base}-update"
+
+    command = [
+        "git",
+        "checkout",
+        "-b",
+        head,
+        f"origin/{base}",
+    ]
+    result = subprocess.run(command, cwd=WORKING_DIR, capture_output=True, text=True)
+    if result.returncode != 0:
+        logger.error(f"Error creating new branch {head}: {result.stderr}")
+        exit(1)
+
+    return head
+
+
+def apply_patch() -> bool:
+    logger.info("Applying patch ...")
+
+    command = [
+        "git",
+        "apply",
+        "--3way",
+        "../diff.patch",
+    ]
+    result = subprocess.run(command, cwd=WORKING_DIR, capture_output=True, text=True)
+    if result.returncode != 0:
+        if "conflicts" in result.stderr:
+            return True
+        else:
+            logger.error(f"Error applying diff: {result.stderr}")
+            exit(1)
+
+
+def commit_diff(head: str):
+    logger.info("Pushing changes ...")
+
+    add = ["git", "add", "."]
+    subprocess.run(add, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    commit = ["git", "commit", "-m", "chore: merging diff"]
+    subprocess.run(commit, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    push = ["git", "push", "-f", "origin", head]
+    result = subprocess.run(push, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    if result.returncode != 0:
+        logger.error(f"Error pushing changes to branch {head}: {result.stderr}")
+        exit(1)
+
+
+def create_pr(repo: str, base: str, head: str, conflict: bool):
+    logger.info("Creating PR ...")
+
+    title = f"chore: merging diff into branch {base}"
+    body = f"Automated change: merging diff into branch {base}"
+
+    if conflict:
+        title = f"{title} CONFLICTS!"
+        body = f"**Conflicts! Need human intervention!**\n\n{body}"
+
+    repo = gh_client.get_repo(repo)
+    pr = repo.create_pull(base=base, head=head, title=title, body=body)
+    if not pr:
+        logger.error("Error creating PR.")
```

Is there any error message here? Wouldn't a failure be an exception? (In which case we can probably just let it bubble up for a script like this)

---

I had a look at the code here, I think the `if not pr` part is not necessary, because `repo.create_pull` would raise an exception if `status >= 400`. I will remove this check and this line of log here.